torch/csrc/jit/codegen/cuda/lower2device.cpp

```diff
@@ -19,11 +20,13 @@ std::vector<Expr*> GPULower::getLoweredExprs() {
   // Validate and make some minor modifications in preparation to generate code.
   PrepareForLowering(fusion_);
 
+  auto preds = ThreadPredicates::compute(fusion_);
```

Comment:
since predicates are common lowering state, instead of passing them around why not making them part of the Fusion itself? (and PrepareForLowering() would initialize them)